runtime/autoload/man.vim

```diff
@@ -121,21 +122,33 @@ function! s:extract_sect_and_name_ref(ref) abort
   return [tolower(split(left[1], ')')[0]), left[0]]
 endfunction
 
-function! s:verify_exists(sect, name) abort
-  let path = system(s:man_cmd.' '.s:man_find_arg.' '.s:man_args(a:sect, a:name))
-  if path !~# '^\/'
-    if empty(a:sect)
-      throw 'no manual entry for '.a:name
-    endif
+function! s:get_path(sect, name) abort
+  if empty(a:sect)
     let path = system(s:man_cmd.' '.s:man_find_arg.' '.shellescape(a:name))
     if path !~# '^\/'
-      throw 'no manual entry for '.a:name.'('.a:sect.') or '.a:name
+      throw 'no manual entry for '.a:name
     endif
+    return path
   endif
-  if a:name =~# '\/'
-    " We do not need to extract the section/name from the path if the name is
-    " just a path.
-    return ['', a:name, path]
+  " The '-s' flag is very useful.
+  " We do not need to worry about stuff like 'printf(echo)'
+  " (two manpages would be interpreted by man without -s)
+  " We do not need to check if the sect starts with '-'
+  " Lastly, the 3pcap section on macOS doesn't work without -s
+  return system(s:man_cmd.' '.s:man_find_arg.' -s '.shellescape(a:sect).' '.shellescape(a:name))
```

Comment:
`shellescape()` wouldn't be necessary if a list were passed to `system()` instead of a string. `s:man_cmd` could be stored as a list instead of a string, then the list can be `extend()`'d here.